lnbits/core/services.py

```diff
@@ -124,16 +124,17 @@ async def pay_invoice(
     will regularly check for the payment.
     """
     invoice = bolt11.decode(payment_request)
+
```

Comment:
- is this just a casual refactoring or is it related to the `bolt11` changes?

Reply:
so bolt11 lib supports amountless invoices, but lnbits does not. so this is casual refactoring to make mypy happy with those optional amounts.

also i moved it before we open a database connection, because we can actually raise before that.